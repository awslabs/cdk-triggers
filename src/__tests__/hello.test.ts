import { ResourcePart, SynthUtils } from '@aws-cdk/assert';
import '@aws-cdk/assert/jest';
import * as lambda from '@aws-cdk/aws-lambda';
import * as sns from '@aws-cdk/aws-sns';
import { Stack } from '@aws-cdk/core';
import * as triggers from '../index';

test('minimal usage', () => {
  // GIVEN
  const stack = new Stack();
  const triggeringResource = new sns.Topic(stack, 'MyTopic');
  const trigger = new lambda.Function(stack, 'MyTriggerHandler', {
    runtime: lambda.Runtime.NODEJS_12_X,
    code: lambda.Code.fromInline('zoo'),
    handler: 'index.handler',
  });

  // WHEN
  new triggers.AfterCreate(stack, 'MyTrigger', {
    handler: trigger,
    resources: [triggeringResource],
  });

  // THEN
  expect(SynthUtils.synthesize(stack).template).toMatchSnapshot();

  expect(stack).toHaveResource('Custom::Trigger', {
    HandlerArn: {
      'Fn::GetAtt': [
        'MyTriggerHandlerD6B1FF23',
        'Arn',
      ],
    },
  });

  expect(stack).toHaveResource('Custom::Trigger', {
    DependsOn: ['MyTopic86869434'],
  }, ResourcePart.CompleteDefinition);
});
