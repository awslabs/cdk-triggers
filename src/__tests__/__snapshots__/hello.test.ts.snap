// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minimal usage 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097ArtifactHashA207A845": Object {
      "Description": "Artifact hash for asset \\"ee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097\\"",
      "Type": "String",
    },
    "AssetParametersee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097S3Bucket8E56AC33": Object {
      "Description": "S3 bucket for asset \\"ee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097\\"",
      "Type": "String",
    },
    "AssetParametersee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097S3VersionKeyB4A96DD3": Object {
      "Description": "S3 key for asset version \\"ee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AWSCDKTriggerCustomResourceProviderCustomResourceProviderHandler97BECD91": Object {
      "DependsOn": Array [
        "AWSCDKTriggerCustomResourceProviderCustomResourceProviderRoleE18FAF0A",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097S3Bucket8E56AC33",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097S3VersionKeyB4A96DD3",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersee94b9804b56d6659a1e72ad0e82bae15d18e2df0c2127f84b42aa935b122097S3VersionKeyB4A96DD3",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "AWSCDKTriggerCustomResourceProviderCustomResourceProviderRoleE18FAF0A",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWSCDKTriggerCustomResourceProviderCustomResourceProviderRoleE18FAF0A": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "lambda:InvokeFunction",
                  ],
                  "Effect": "Allow",
                  "Resource": Array [
                    "*",
                  ],
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MyTopic86869434": Object {
      "Type": "AWS::SNS::Topic",
    },
    "MyTriggerB6CCCACE": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "MyTopic86869434",
      ],
      "Properties": Object {
        "HandlerArn": Object {
          "Fn::GetAtt": Array [
            "MyTriggerHandlerD6B1FF23",
            "Arn",
          ],
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWSCDKTriggerCustomResourceProviderCustomResourceProviderHandler97BECD91",
            "Arn",
          ],
        },
      },
      "Type": "Custom::Trigger",
      "UpdateReplacePolicy": "Delete",
    },
    "MyTriggerHandlerD6B1FF23": Object {
      "DependsOn": Array [
        "MyTriggerHandlerServiceRoleFC0CFFAB",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "zoo",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MyTriggerHandlerServiceRoleFC0CFFAB",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyTriggerHandlerServiceRoleFC0CFFAB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
